lib/emque/producing/message/message.rb

```diff
@@ -130,16 +128,24 @@ def to_json
         Oj.dump(data, :mode => :compat)
       end
 
-      def publish(publisher=nil)
-        publisher ||= Emque::Producing.publisher
+      def publish_all
+        publish(publishers: Emque::Producing.publishers.keys)
+      end
+
+      def publish(publishers:)
         log "publishing...", true
         if valid?
           log "valid...", true
           if Emque::Producing.configuration.publish_messages
             message = process_middleware(to_json)
-            sent = publisher.publish(topic, message_type, message, partition_key, raise_on_failure?)
-            log "sent #{sent}"
-            raise MessagesNotSentError.new unless sent
+            publishers.each do |publisher|
```

Comment:
need to figure out failure in the loop. If first publisher fails, should second?